hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -360,8 +360,8 @@ private ReplicationSourceInterface createSource(String queueId, ReplicationPeer
     MetricsSource metrics = new MetricsSource(queueId);
     sourceMetrics.put(queueId, metrics);
     // init replication source
-    src.init(conf, fs, logDir, this, queueStorage, replicationPeer, server, queueId, clusterId,
-      walFileLengthProvider, metrics);
+    src.init(conf, fs, logDir, this, queueStorage, replicationPeer, server, server.getServerName(),
```

Comment:
There will be a time when the passed in 'server' differs from server.getServerName?

Reply:
Perhaps say more about this?

Reply:
* @param server the server which start and run this replication source
   * @param producer the name of region server which produce WAL to the replication queue

Is this comment clearly?

Reply:
One replication queue can be identified by <Producer ServerName, QueueId>.

Reply:
And one ReplicationSource is responsible for one replication queue.

Reply:
When replication offload disabled, the ReplicationSource will be started in RegionServer inside. So the producer name is same with the Server's name.